samples/client/petstore/elixir/lib/openapi_petstore/request_builder.ex

```diff
@@ -110,20 +110,22 @@ defmodule OpenapiPetstore.RequestBuilder do
 
   ## Parameters
 
-  - arg1 (Tesla.Env.t | term) - The response object
+  - arg1 ({:ok, Tesla.Env.t} | term) - The response object
   - arg2 (:false | struct | [struct]) - The shape of the struct to deserialize into
 
   ## Returns
 
   {:ok, struct} on success
   {:error, term} on failure
   """
-  @spec decode(Tesla.Env.t | term()) :: {:ok, struct()} | {:error, Tesla.Env.t} | {:error, term()}
-  def decode(%Tesla.Env{status: 200, body: body}), do: Poison.decode(body)
+  @spec decode({:ok, Tesla.Env.t} | term()) :: {:ok, struct()} | {:error, Tesla.Env.t} | {:error, term()}
+  def decode({:ok, %Tesla.Env{status: 200, body: body}}), do: Poison.decode(body)
   def decode(response), do: {:error, response}
```

Comment:
The current `response` matches on all responses whicht didn't result in a 200 status code (which might needs to be fixed in another Issue - I'll open that one later)

Therefore I would suggest something like this:
```suggestion
   def decode({:ok, %Tesla.Env{} = response), do: {:error, response}
```